cmd/launcher/launcher.go

```diff
@@ -144,6 +146,18 @@ func runLauncher(ctx context.Context, cancel func(), opts *launcher.Options) err
 	flagController := flags.NewFlagController(logger, stores[storage.AgentFlagsStore], fcOpts...)
 	k := knapsack.New(stores, flagController, db)
 
+	// Need to set up the log shipper so that we can get the logger early
+	// and pass it to the various systems.
+	var logShipper *logshipper.LogShipper
+	if k.ControlServerURL() != "" {
+		logShipper, err = logshipper.New(k)
```

Comment:
How does this work when log shipping is disabled?

Reply:
updated so we'll only start shipping if we have a host.

Reply:
So to be clear...

Can we merge this today and have no effect until we set this from the control server?

Reply:
I missed this comment earlier, I went back and introduced a new "LogShippingEnabled" flag. So the log shipper can be set up and ready to go until we flip the flag. It will still "run" in the background, but just ignore calls to `Log()` until it's enabled. So we can merge it and it will have not effect until we set it from the control server.

Reply:
dropped the enabled flag, now just enabling based on log_ingest_url presence

Reply:
Sorry for resolving/unresolving this comment thread like 3 times, this was a little confusing to me. Maybe this section belongs inside the `if k.ControlServerURL() == "" { ... } else { ...` block later, where all the other consumers/subscribers are?

Reply:
Ah sorry, I missed the comment explaining why it has to happen here -- I think I just got confused toggling between this comment thread/references to older versions of this code/logshipper.go, and I don't think this is actually confusing or requires comment updates or anything.